cpp/unreal_plugins/SimulationController/Source/SimulationController/CameraSensor.cpp

```diff
@@ -91,41 +91,40 @@ CameraSensor::CameraSensor(
-            render_pass_desc.scene_capture_component_->PostProcessSettings.bOverride_DynamicGlobalIlluminationMethod = true;
-            render_pass_desc.scene_capture_component_->PostProcessSettings.DynamicGlobalIlluminationMethod           = EDynamicGlobalIlluminationMethod::Lumen;
-            render_pass_desc.scene_capture_component_->PostProcessSettings.bOverride_ReflectionMethod = true;
-            render_pass_desc.scene_capture_component_->PostProcessSettings.ReflectionMethod           = EReflectionMethod::Lumen;
-            render_pass_desc.scene_capture_component_->PostProcessSettings.bOverride_LumenSurfaceCacheResolution = true;
-            render_pass_desc.scene_capture_component_->PostProcessSettings.LumenSurfaceCacheResolution           = 1.0f;
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.bOverride_DynamicGlobalIlluminationMethod = true;
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.DynamicGlobalIlluminationMethod           = EDynamicGlobalIlluminationMethod::Lumen;
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.bOverride_ReflectionMethod = true;
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.ReflectionMethod           = EReflectionMethod::Lumen;
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.bOverride_LumenSurfaceCacheResolution = true;
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.LumenSurfaceCacheResolution           = 1.0f;
         } else {
-            // TODO: turn off as many rendering features as possible for efficiency
-            UMaterial* material = LoadObject<UMaterial>(nullptr, *Unreal::toFString(RENDER_PASS_MATERIAL.at(render_pass_name)));
+            // TODO (MR): turn off as many rendering features as possible for efficiency
+            auto material = LoadObject<UMaterial>(nullptr, *Unreal::toFString(RENDER_PASS_MATERIAL.at(render_pass_name)));
             SP_ASSERT(material);
-            render_pass_desc.scene_capture_component_->PostProcessSettings.AddBlendable(UMaterialInstanceDynamic::Create(material, parent_actor_), 1.0f);
+            render_pass_desc.scene_capture_component_2d_->PostProcessSettings.AddBlendable(UMaterialInstanceDynamic::Create(material, actor_), 1.0f);
         }
 
-        render_pass_desc.scene_capture_component_->AttachToComponent(camera_component, FAttachmentTransformRules::SnapToTargetNotIncludingScale);
-        render_pass_desc.scene_capture_component_->SetVisibility(true);
-        render_pass_desc.scene_capture_component_->RegisterComponent();
+        render_pass_desc.scene_capture_component_2d_->SetupAttachment(camera_component);
+        render_pass_desc.scene_capture_component_2d_->SetVisibility(true);
+        render_pass_desc.scene_capture_component_2d_->RegisterComponent();
```

Comment:
Move `RegisterComponent` to be the first function that gets called after `SP_ASSERT(render_pass_desc.scene_capture_component_2d_)`

Reply:
Updated code